pil-analyzer/src/statement_processor.rs

```diff
@@ -259,6 +272,20 @@ where
         }
     }
 
+    /// Handles a type occurring in a context that has the given type variables declared.
+    fn handle_type(&self, type_vars: &HashSet<&String>, ty: &Type<parsed::Expression>) -> Type {
+        let mut ty = self.evaluate_array_lengths(ty.clone())
+            .map_err(|e| panic!("Error evaluating expressions in type name \"{}\" to reduce it to a type:\n{e})", ty))
+            .unwrap();
+        ty.map_to_type_vars(type_vars);
+        ty.contained_named_types_mut().for_each(|n| {
+            let name = self.driver.resolve_ref(n);
+            // TOOD check that it actually is a type.
```

Comment:
still to do

Reply:
I think I would like to take this out of this PR. Created issue: https://github.com/powdr-labs/powdr/issues/1166